### DISCUSSION TITLE
Popup on shell error

### DISCUSSION THREAD

**Discussion by Marsevil** [helix-editor/helix#6917] [Ideas]
I'm wondering why executing a shell command through helix `:sh my_command` show the results if the function success but not if the function failed.
In my case, I've setup a custom key shortcut to run `make` and I can't see the problem if the command fails.

**Comment by mweihs** [comment]
Failure might be output to stderr? Try `:sh my_command 2>&1`

**Reply by Marsevil** [reply]
Thx for the answer. I've tested, but, my point is that if the command fail the message is shown in the status bar but not inside a popup. Maybe there is a way to open the status bar message buffer in a popup ??

**Reply by OlafLostViking** [reply]
Hi!

I just stumbled into the same problem: Wanting to see the output of a build step. Funnily enough, my first try was indeed the `2>&1` approach. But that lead to only showing the return code in the status bar instead of the output that's cramped into the line without the redirection.

Did you, @Marsevil , find a solution to have a popup from failed shell commands?

Thanks!

**Reply by Marsevil** [reply]
Hi, no news here, I now prefer to avoid using `sh` in helix and use `Ctrl+Z`/`fg` and/or multi-tab to run my commands

**Reply by arut-plus** [reply]
Try `:sh some_command >&2`, @Marsevil.
Works for me for the commands `go run`, `go test`, etc.
Without `>&2` I just get a `Shell command failed: status 1` without any output of the commands themselves.
